packages/worker/src/worker.ts

```diff
@@ -451,35 +456,67 @@ export class Worker {
    * @throws {@link DeterminismViolationError} if the workflow code is not compatible with the history.
    */
   public static async runReplayHistory(options: ReplayWorkerOptions, history: History | unknown): Promise<void> {
+    const hist = this.validateHistory(history);
+    const [worker, pusher] = await this.constructReplayWorker(options);
+    const rt = Runtime.instance();
+    await rt.pushHistory(pusher, 'fakeid', hist);
+    rt.closeHistoryStream(pusher);
+    await this.runReplayerToCompletion(worker, {});
+  }
+
+  /**
+   * Create a replay Worker, running all histories provided by the passed in iterable.
+   * If `options.failFast` is set, the function will throw upon the first encountered error.
+   * If not, all histories will be replayed and the returned results object will contain information
+   * about any failures.
+   *
+   * @throws {@link DeterminismViolationError} if in fail fast mode and there is some incompatible
+   * history.
+   */
+  public static async runReplayHistories(
+    options: ReplayWorkerOptions,
+    histories: AsyncIterable<HistoryAndWorkflowID>
```

Comment:
maybe:
```suggestion
    options: ReplayHistoriesWorkerOptions,
```

```
type ReplayHistoriesWorkerOptions = ReplayWorkerOptions & {
  failFast: 
  histories: AsyncIterable<HistoryAndWorkflowID>
```

Reply:
`runReplayHistory` explicitly chose to separate options from the history provided, so this is consistent with that